lib/utils/deploy-legacy.coffee

```diff
@@ -0,0 +1,145 @@
+Promise = require('bluebird')
+
+getBuilderPushEndpoint = (baseUrl, owner, app) ->
+	querystring = require('querystring')
+	args = querystring.stringify({ owner, app })
+	"https://builder.#{baseUrl}/v1/push?#{args}"
+
+getBuilderLogPushEndpoint = (baseUrl, buildId, owner, app) ->
+	querystring = require('querystring')
+	args = querystring.stringify({ owner, app, buildId })
+	"https://builder.#{baseUrl}/v1/pushLogs?#{args}"
+
+bufferImage = (docker, imageId, bufferFile) ->
+	Promise = require('bluebird')
+	streamUtils = require('./streams')
+
+	image = docker.getImage(imageId)
+	imageMetadata = image.inspect()
+
+	Promise.join image.get(), imageMetadata.get('Size'), (imageStream, imageSize) ->
+		streamUtils.buffer(imageStream, bufferFile)
+		.tap (bufferedStream) ->
+			bufferedStream.length = imageSize
+
+showPushProgress = (message) ->
+	visuals = require('resin-cli-visuals')
+	progressBar = new visuals.Progress(message)
+	progressBar.update({ percentage: 0 })
+	return progressBar
+
+forwardStatusMessageFromRemote = (logger, msg) ->
+	msg.split(/\r\n|\n/).forEach (line) ->
+		if /^Warning: This endpoint is deprecated/.test(line)
+			return
+		logger.logInfo(line)
+
+uploadToPromise = (uploadRequest, logger) ->
+	new Promise (resolve, reject) ->
+
+		handleMessage = (data) ->
+			data = data.toString()
+			logger.logDebug("Received data: #{data}")
+
+			try
+				obj = JSON.parse(data)
+			catch e
+				logger.logError('Error parsing reply from remote side')
+				reject(e)
+				return
+
+			if obj.type?
+				switch obj.type
+					when 'error' then reject(new Error("Remote error: #{obj.error}"))
+					when 'success' then resolve(obj)
+					when 'status' then forwardStatusMessageFromRemote(logger, obj.message)
+					else reject(new Error("Received unexpected reply from remote: #{data}"))
+			else
+				reject(new Error("Received unexpected reply from remote: #{data}"))
```

Comment:
Nit: Does this need to be a special case? I would've thought `obj.type = undefined` would've hit the else case above anyway.

Reply:
Ah, you're right, I merely reverted the removal of the old code making only the needed changes in order to work. Will change this.

Reply:
In fact I don't think it should even raise -- what's wrong with receiving messages it doesn't know? -- the exception prevents extending the builder push endpoint. But on the other CLI v6 does that as well, so I'm not sure if it makes sense stopping raising.

Reply:
Yeah, good point. I think sticking with reject is fine for now, since it's consistent with before and it doesn't limit us any more than we already are, but yeah if we were building this again in future we might well want to do differently.